lib/pbench/test/unit/agent/test_tool_group.py

```diff
@@ -0,0 +1,147 @@
+import os
+import pytest
+
+from pathlib import Path
+
+from pbench.agent.tool_group import BadToolGroup, ToolGroup
+
+# from pbench.common.exceptions import BadConfig
+
+
+class TestToolGroup:
+
+    pbench_run = "/mock/pbench_run"
+    group = "tool-group"
+    mock_tg_dir_name = Path(pbench_run, f"tools-v1-{group}")
+
+    def mock_resolve(self: Path, strict: bool = False):
+        """
+        Return a "full" fake file name
+
+        Args:
+            strict: Not used in the mock
+
+        Returns:
+            new Path with fake resolution
+        """
+        return Path("/mock/pbench-agent/")
```

Comment:
Not sure what @dbutenhof means by `return self`, but I think you might want to create a single function, or fixture, instead of four individual mock methods, where in the fixture you provide the same environment used for at least the last 4 tests in the file.

That would reduce this code substantially and make it easier to understand what the mocked environment looks like (where you can document the resulting mocked environment in the docstring of the method or fixture).

Reply:
What I meant is that the mock is always returning "expected" data regardless of the input (the `Path` on which the mock is triggered). Since this is part of the test, I think it should at least be asserting that the input (`self: Path`) has the expected value; returning the expected string without checking could hide an error in the CUT.

Reply:
Since I composed it long before the previous two comments, [my response](https://github.com/distributed-system-analysis/pbench/pull/2832#discussion_r871657133) appears above them...just didn't want folks to miss it.  🥴